### PR TITLE
Remove HTML within schema

### PR DESCRIPTION
**Pull request by MIslam92**
### What is the context of this PR?
Some schemas contain markup with opening tags which aren't closed. We should remove this in favour of using the content object within the schema instead

Trello: https://trello.com/c/7Pqn9TmB/1936-remove-html-within-schemas-xs

### How to review 

1) Review this PR first: https://github.com/ONSdigital/eq-schema-validator/pull/45

2) Check that from the surveys listed in the card, all structural embedded HTML is removed form our schemas and it is still displayed as it previously did.

### CODE REVIEW

**Review by shahi645 — CHANGES_REQUESTED** [review]
As question content is a new thing, needs to be defined in schema validator

File: data/en/1_0005.json
Comment (davecharles) [inline data/en/1_0005.json]: Is the extra space (between `of` and `weekly`) relevant?

File: data/en/1_0005.json
Comment (davecharles) [inline data/en/1_0005.json]: Space: `change in..{{respondent`

File: data/en/1_0005.json
Comment (davecharles) [inline data/en/1_0005.json]: Space: `change in..{{respondent`

File: data/en/1_0005.json
Comment (davecharles) [inline data/en/1_0005.json]: Space: `change in..{{respondent`

File: data/en/1_0005.json
Comment (davecharles) [inline data/en/1_0005.json]: This looks wrong: `...paid five weekly monthly`

File: data/en/1_0005.json
Comment (davecharles) [inline data/en/1_0005.json]: Space: `change in..{{respondent`

File: data/en/e_commerce.json
Comment (davecharles) [inline data/en/e_commerce.json]: `(VPN)` not `(VPN)}`

File: data/en/e_commerce.json
Comment (davecharles) [inline data/en/e_commerce.json]: I don't know when `yarn format` was introduced and if applying it would have obfuscated your changes.  Whichever, maybe use a consistent line continuation style.

**Review by davecharles — CHANGES_REQUESTED** [review]
These also have unnecessary html.  Understand most are test schemas but we should be consistent:
- 1_0215.json
- census_communal.json
- census_household.json
- mbs_0111.json
- test_0102.json
- test_0112.json
- test_0205.json
- test_0213.json
- test_introduction.json
- test_radio_checkbox_descriptions.json
- test_repeating_household.json
- test_relationship_household.json
- test_question_guidance.json

**Review by shahi645 — CHANGES_REQUESTED** [review]
We need to add question content to the other question type templates too

**Comment by MebinAbraham** [comment]
Looks good. Happy to approve once validator PR is in master and Travis passes.

**Review by shahi645 — APPROVED** [review]
Looks good, needs front end review. Also would be good to put these schemas through browser stack maybe, I've checked them out on safari and chrome multiple devices. 👍

**Review by hamishtaplin — CHANGES_REQUESTED** [review]
Is it me or does this result in all the `p` tags being removed and replaced with `div`?

It should keep the `p` tags, really.

**Comment by hamishtaplin** [comment]
Definitely should keep the `p` tags:

screenshot 2018-04-18 15 11 28

**Comment by WickyNilliams** [comment]
How does this change affect content from author/publisher?

**Review by nealedj — CHANGES_REQUESTED** [review]
8 schemas still contain `br` tags. These are all question descriptions.

In all these cases I think they would be better defined in the content as separate description lines.

It's worth mentioning that the screenshot @hamishtaplin showing condensed text is what the application was already doing before this PR. But it is pretty unreadable.

**Comment by nealedj** [comment]
@WickyNilliams it won't break anything as this change will still allow question descriptions to contain markup. But we should have a chat about how to get author defining description paragraphs in the `content.description` attribute rather than the `description` attribute. I'll add it for discussion at the next tech session.

**Comment by WickyNilliams** [comment]
@nealedj publisher previously assumed runner would wrap content in `<p>` tags. As this is no longer the case, should publisher start wrapping content in `<p>` tags (or rather, not removing them, as everything author produces is already wrapped in a `<p>`)?

Even if it won't break, will it start looking compact, as shown by Hamish?

**Comment by nealedj** [comment]
@WickyNilliams 

What @hamishtaplin showed was a page from the census schema which was using `<br>` tags - that's why it looked broken. It's not related to any of the changes in this PR though - it never had any <p> tags in that description.

Wrapping everything in the description in a <p> tag doesn't appear to have any effect on the layout.

However, we are repeating a div with the `question__description` class which seems redundant... I'll request a change for that

File: app/templates/partials/question.html
Comment (nealedj) [inline app/templates/partials/question.html]: I think it might be more semantic to iterate some `<p>` tags within the description `<div>`. Each line of the description belongs to the description, it isn't a separate description.

E.g.
```
<div class="question__description mars">
  {{question.description}}
  {% for content in question.content %}
    <p>{{content.description}}</p>
  {% endfor %}
</div>
```

**Review by nealedj — CHANGES_REQUESTED** [review]

File: app/templates/partials/question.html
Comment (nealedj) [inline app/templates/partials/question.html]: That's not what I meant - that's still rendering a div for every line of the description:

Currently it's this:
```
<div class="question__description mars">
      {{question.description}}
    {% for content in question.content %}
        <div class="question__description mars">
            <p>{{content.description}}</p>
        </div>
    {% endfor %}
    </div>
```

But I think it'd be better as this:
```
<div class="question__description mars">
  {{question.description}}
  {% for content in question.content %}
    <p>{{content.description}}</p>
  {% endfor %}
</div>
```

**Review by nealedj — CHANGES_REQUESTED** [review]
1_0213.json is failing due to invalid JSON also.

File: scripts/test_schemas.sh
Comment (MIslam92) [inline scripts/test_schemas.sh]: Will change back once PR approved and ready to merge

**Review by nealedj — APPROVED** [review]

**Comment by MIslam92** [comment]
@hamishtaplin @davecharles Whenever your free could you take another look at this to check if your comments have been addressed?

**Review by davecharles — APPROVED** [review]

**Review by nealedj — CHANGES_REQUESTED** [review]
As discussed - the current structure of the content item could be simplified as below:

```
"content": {
    "text": [
        "You have completed the previous section.",
        "<em>Cloud computing</em>"
    ],
    "list": [
        "Are delivered from <em>servers</em> of service providers",
        "Can be easily <em>scaled</em> up or down (for example; the number of users or change of storage capacity)",
        "Can be used <em>on-demand by the user</em>, at least after the initial set up (without human interaction with the service provider)",
        "Are <em>paid</em> for, either by the user, by the capacity used or they are pre-paid"
    ],
    "text": [
        "Cloud computing may include connections via Virtual Private Networks (VPN)"
    ]
}
```

**Pull request closed without merging**
